ath79: mr18: use nvmem for MACs

neheb opened the pull request:
NVMEM in UBI support has been present for a while. Use it to get the MACs. Seems caldata needs to continue to be obtained through userspace.

ping @Leo-PL @chunkeey 

I can't say I understand the nvmem comments. sometimes caldata is in ubi and sometimes not?

Leo-PL commented:
I'll try to test it in the coming days. Caldata should be accessible via NVMEM too, but I need to check - I've built a test firmware with that (made the changes separately by myself) but haven't tested it yet.

neheb commented:
I converted the cal data to dts. I kept the weird odm-caldata handling. I still don't understand why it's there or if it's even needed.

Leo-PL commented:
Some early devices had caldata volume missing from UBI partition, and only odm-caldata was available. At the same time, sometimes that used different ECC algorithm, causing failed reads. Ping @chunkeey for more details.

neheb committed:
ath79: mr18: use nvmem for MACs

NVMEM in UBI support has been present for a while. Use it to get the
MACs. Seems caldata needs to continue to be obtained through userspace.

This fixes label-mac-device as it needs a mac-address nvmem cell.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17068
Signed-off-by: Robert Marko <robimarko@gmail.com>

neheb commented:
ping @hauke
